frontend/src/slices/auth/actions.ts

```diff
@@ -18,4 +20,21 @@ const signUp = createAsyncThunk<
   return authApi.signUp(registerPayload);
 });
 
-export { signUp };
+const signIn = createAsyncThunk<
+  UserDetailsDto,
+  UserSignInRequestDto,
+  AsyncThunkConfig
+>(`${sliceName}/sign-in`, async (loginPayload, { rejectWithValue, extra }) => {
+  const { authApi, storage } = extra;
+
+  try {
+    const userData = await authApi.signIn(loginPayload);
+    await storage.set('token', userData.token);
+
+    return userData.user;
+  } catch (error) {
+    return rejectWithValue(error as Error);
+  }
```

Comment:
No need for try catch block. Action should automatically throw if there's an error